lib/parse-number.c

```diff
@@ -146,6 +147,12 @@ _process_suffix(const gchar *suffix, gint64 *d)
   return TRUE;
 }
 
+static gboolean
+_no_suffix(const gchar *suffix)
+{
+  return (*suffix == '\0' || isspace(*suffix));
```

Comment:
Can you explain when the `isspace` branch is possible?

Reply:
The previous implementation was faulty. The parsing only checked if the `endptr` is equal to the first character in the string representation of the number, but it is possible that the `strtoll` fails on other point and not the start. For example, if `strtoll` is called with base detection and the following number is supplied: `018`.

From `strtoll` point of view, the number is `1`, and the `endptr` is `8`, but that number is clearly not valid, the previous implementation marked it as a valid number, ignoring the `8`.

I thought I had to define a number in a way, that it is followed by something. I came up with `\0` and whitespace.